app/routes.py

```diff
@@ -1,2 +1,151 @@
-from flask import Blueprint
+from flask import Blueprint, jsonify, abort, make_response, request
+from app.models.planet import Planet
+from app import db
 
+planets_bp = Blueprint("planets_bp", __name__, url_prefix = "/planets")
+
+# HELPER FUNCTION
+def validate_id(class_obj, id):
+    try:
+        object_id = int(id)
+    except:
+        abort(make_response({"message":f"{class_obj} {id} is invalid"}, 400))
+    
+    query_result = class_obj.query.get(object_id)
+    if not query_result:
+        abort(make_response({"message":f"{class_obj} {id} is not found"}, 404))
+
+    return query_result
+
+# CREATE RESOURCE
+@planets_bp.route("", methods = ["POST"])
+def create_planet():
+    request_body = request.get_json()
+    
+    new_planet = Planet.from_json(request_body)
+        
+    db.session.add(new_planet)
+    db.session.commit()
+
+    return make_response(f"Planet {new_planet.name} has been created", 201)
+
+
+# GET ALL RESOURCES
+@planets_bp.route("", methods = ["GET"])
+def get_all_planets():
+    results_list = []
+
+    color_param= request.args.get("color")
+    description_param= request.args.get("description")
+    
+    if color_param:
+        planets = Planet.query.filter_by(color=color_param)
+    elif description_param:
+        planets = Planet.query.filter_by(description=description_param)
+    else:
+        planets = Planet.query.all()
+
+    for planet in planets:
+        results_list.append(planet.to_dict())
+    
+    return jsonify(results_list), 200
+
+
+# GET ALL RESOURCES
+@planets_bp.route("/<planet_id>", methods = ["GET"])
+def get_one_planet(planet_id):
+    planet = validate_id(Planet, planet_id)
+
+    return jsonify(planet.to_dict())
+
+
+# UPDATE RESOURCE
+@planets_bp.route("/<planet_id>", methods = ["PUT"])
+def update_planet(planet_id):
+    planet = validate_id(Planet, planet_id)
+
+    request_body = request.get_json()
+
+    try:
+        planet.update(request_body)
+    except KeyError as error:
+        return make_response({'message': f"Missing attribute: {error}"}, 400)
```

Comment:
This is good validation, something similar could be done in the post request.   It should also be tested